config/src/config/consensus_config.rs

```diff
@@ -36,9 +36,22 @@ pub struct ConsensusConfig {
     // Decides how long the leader waits before proposing empty block if there's no txns in mempool
     // the period = (poll_count - 1) * 30ms
     pub quorum_store_poll_count: u64,
+    // Whether to create partial blocks when few transactions exist, or empty blocks when there is
+    // pending ordering, or to wait for quorum_store_poll_count * 30ms to collect transactions for a block
+    //
+    // It is more efficient to execute larger blocks, as it creates less overhead. On the other hand
+    // waiting increases latency (unless we are under high load that added waiting latency
+    // is compensated by faster execution time). So we want to balance the two, by waiting only
+    // when we are saturating the execution pipeline:
+    // - if there are more pending blocks then usual in the execution pipeline,
+    //   block is going to wait there anyways, so we can wait to create a bigger/more efificent block
+    // - in case our node is faster than others, and we don't have many pending blocks,
+    //   but we still see very large recent (pending) blocks, we know that there is demand
+    //   and others are creating large blocks, so we can wait as well.
+    pub wait_for_full_blocks_above_num_pending_uncommitted_blocks: usize,
```

Comment:
How about the names? Can we shorten it? May be `wait_until_pending_blocks` and `wait_above_fill_threshold` ?

Reply:
I prefer names that explain what they are about. especially for configurations, that can have undesirable effects if missunderstood.

wait_until_pending_blocks / wait_above_fill_threshold doesn't say what wait it is referring to. also pending blocks is not clear "pending" how, but I can shorten it to 

wait_for_full_blocks_above_pending_blocks

and keep the other the same?